pkg/devfile/parser/context/context.go

```diff
@@ -38,6 +38,29 @@ func NewDevfileCtx(path string) DevfileCtx {
 	}
 }
 
+// PopulateDevfile fills the DevfileCtx struct with relevant context info
+func (d *DevfileCtx) PopulateDevfile() (err error) {
+
+	// Get devfile APIVersion
+	if err := d.SetDevfileAPIVersion(); err != nil {
+		return err
+	}
+
+	// Check if the apiVersion is supported
+	if !d.IsApiVersionSupported() {
+		return fmt.Errorf("devfile apiVersion '%s' not supported in odo", d.apiVersion)
+	}
+	klog.V(4).Infof("devfile apiVersion '%s' is supported in odo", d.apiVersion)
+
+	// Read and save devfile JSON schema for provided apiVersion
+	if err := d.SetDevfileJSONSchema(); err != nil {
```

Comment:
This and the following lines could be simplified to simply `return d.SetDevfileJSONSchema()`

Reply:
Done